packages/ember-glimmer/tests/integration/syntax/with-test.js

```diff
@@ -30,4 +30,66 @@ moduleFor('Syntax test: {{#with}}', class extends SharedConditionalsTest {
 
     this.assertText('False');
   }
+
+  ['@test updating the context should update the alias']() {
+    this.render(`{{#with cond1 as |cond|}}{{cond.greeting}}{{else}}False{{/with}}`, {
+      cond1: { greeting: 'Hello' }
+    });
+
+    this.assertText('Hello');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Hello');
+
+    this.runTask(() => set(this.context, 'cond1', { greeting: 'Hola'}));
+
+    this.assertText('Hola');
+
```

Comment:
Should keep the initial -> no-op -> mutation -> replacement pattern

mutation would be setting `cond1.greeting` to Hola

replacement would be setting cond1 to a new object `{ greeting: "Hello" }`